protogen/codegen.go

```diff
@@ -0,0 +1,257 @@
+package protogen
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/openconfig/ygot/util"
+	"github.com/openconfig/ygot/ygen"
+)
+
+// ProtoCodeGenerator is a structure that is used to pass arguments as to
+// how the output protobuf code should be generated.
+type ProtoCodeGenerator struct {
+	// Config stores the configuration parameters used for code generation.
+	Config ygen.GeneratorConfig
+	// ProtoOptions stores a struct which contains Protobuf specific options.
+	ProtoOptions ProtoOpts
```

Comment:
```suggestion
	Options Opts
```

Reply:
Proposal to address this here: https://github.com/openconfig/ygot/pull/706 PTAL